File: backend/app/routers/budget.py
from fastapi import APIRouter
from app.schemas.budget import BudgetCalculateRequest, BudgetBreakdownResponse
from app.services.budget_calculator import calculate_detailed_budget

router = APIRouter(prefix="/api/budget", tags=["Budget"])

@router.post("/calculate", response_model=BudgetBreakdownResponse)
def calculate_budget(req: BudgetCalculateRequest):
    return calculate_detailed_budget(
        destination_name=req.destination_name,
        starting_city=req.starting_city,
        duration_days=req.duration_days,
        travelers_count=req.travelers_count,
        transportation=req.transportation,
        accommodation=req.accommodation,
        trip_pace=req.trip_pace
    )

File: frontend/src/api/budget.ts
import { apiClient } from './client';
import { BudgetBreakdown } from '../types';

export interface CalculateBudgetPayload {
  destination_name: string;
  starting_city: string;
  duration_days: number;
  travelers_count: number;
  transportation: string;
  accommodation: string;
  trip_pace: string;
}

export const calculateBudget = async (payload: CalculateBudgetPayload): Promise<BudgetBreakdown> => {
  const res = await apiClient.post<BudgetBreakdown>('/api/budget/calculate', payload);
  return res.data;
};
